magento/magento2
# save() in customerRepository runs two times when create the customer with subcription

### Description (*)


 In the CreatePost class, now, it saved 2 times if subscribe = TRUE

```
$customer = $this->accountManagement
                ->createAccount($customer, $password, $redirectUrl);
```
```
            if ($this->getRequest()->getParam('is_subscribed', false)) {
                $extensionAttributes = $customer->getExtensionAttributes();
                $extensionAttributes->setIsSubscribed(true);
                $customer->setExtensionAttributes($extensionAttributes);
                $this->customerRepository->save($customer);
            }
```

**Solution:** We need to save only one time by change the code to :

```
            $extensionAttributes = $customer->getExtensionAttributes();
            $extensionAttributes->setIsSubscribed($this->getRequest()->getParam('is_subscribed', false));
            $customer->setExtensionAttributes($extensionAttributes);

            $customer = $this->accountManagement
                ->createAccount($customer, $password, $redirectUrl);
```


### Fixed Issues (if relevant)


### Manual testing scenarios (*)


Check the save() function in customerRepository should run 1 times instead of run 2 times.

### Questions or comments


### Contribution checklist (*)
